src/parser/statement.js

```diff
@@ -688,7 +686,11 @@ pp.parseClassBody = function (node) {
 
     let isAsyncMethod = !this.match(tt.parenL) && !method.computed && method.key.type === "Identifier" && method.key.name === "async";
     if (isAsyncMethod) {
-      if (this.hasPlugin("asyncGenerators") && this.eat(tt.star)) isGenerator = true;
+      if (this.match(tt.star)) {
+        this.expectPlugin("asyncGenerators");
+        this.next();
```

Comment:
We have this pattern quite often `if (match) expectPlugin(), next()` So maybe we could make this easier in future by doing:
```
if (this.expectPluginIfMatch("plugin", tt.start)) {
 is = true
}
```
What do you think?
But should do that anyway later, to not make this PR even more bigger.